plugins/polio/api/campaigns/subactivities.py

```diff
@@ -0,0 +1,121 @@
+from django.shortcuts import get_object_or_404
+from django_filters.rest_framework import DjangoFilterBackend  # type: ignore
+from rest_framework import permissions, serializers
+
+from iaso.api.common import ModelViewSet
+from iaso.models import Group
+from plugins.polio.api.shared_serializers import GroupSerializer
+from plugins.polio.models import Round, SubActivity, SubActivityScope
+
+
+class SubActivityScopeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = SubActivityScope
+        fields = ["group", "vaccine"]
+
+    group = GroupSerializer(required=False)
+
+
+class SubActivityCreateUpdateSerializer(serializers.ModelSerializer):
+    round_number = serializers.IntegerField(write_only=True, required=False)
+    campaign = serializers.CharField(write_only=True, required=False)
+    scopes = SubActivityScopeSerializer(many=True, required=False)
```

Comment:
shouldn't the scopes be mandatory?

Reply:
I think not, you could have first just the encoding of dates and reason of activity and the scope coming later